jerry-main/main-unix.c

```diff
@@ -283,7 +283,7 @@ wait_for_source_callback (const jerry_char_t *resource_name_p, /**< resource nam
                           size_t resource_name_size, /**< size of resource name */
                           const jerry_char_t *source_p, /**< source code */
                           size_t source_size, /**< source code size */
-                          void *user_p __attribute__((unused))) /**< user pointer */
+                          void *user_p __attr_unused___) /**< user pointer */
```

Comment:
this file should not rely anything internal

Reply:
Sounds like a bad design.

Reply:
I'm not sure about any of the comments here. How about writing full sentences, ideas, justifications, explanations?

@lygstate: `__attr_unused___` does _not_ come from jerryscript-port.h. That public header writes out all attributes in full, e.g., `__attribute__((noreturn))`. Moreover, it does not make use of the `unused` attribute, neither in its full form nor in the shorthand notation. What's even more strange is that no core part of JerryScript uses the `__attr_unused___` macro, only an internal header of jerry-libc does, which can not possibly leak out to main-unix.c.

@zherczeg: There is no such bad design in jerryscript-port.h. Which can be verified by taking just a bit closer look at the header.

Reply:
I added it into jerryscript-port.h, cause Visual Studio doesnn't suport for __attribute__((noreturn)) or something alike, cause they are porting code, so I moved into jerryscript-port.h @akosthekiss .
Any better idea I appreciate.

Reply:
I worried that the macro was exposed, but it is added in this patch, which should not. I am happy!

Reply:
@zherczeg Help me have a look at the CI, why it's still failed.

Reply:
Fixed.